### PR TITLE
Settings/Manager: Fix - User editor with access to edit content role,…

### PR DESCRIPTION
**Pull request by iNewLegend**
… cannot edit settings (Fix PT#643).

## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

**Comment by iNewLegend** [comment]
TODO: Tests

**Review by kobizz — DISMISSED** [review]

File: assets/dev/js/editor/components/settings/base/manager.js [hunk 1]
```diff
@@ -213,13 +213,13 @@ module.exports = elementorModules.ViewModule.extend( {
...
 
-	onElementorDocumentLoaded: function() {
+	onElementorDocumentLoaded: function( document ) {
 		this.updateStylesheet();
 
 		this.addPanelPage();
 
 		if ( ! elementor.userCan( 'design' ) ) {
-			$e.route( 'panel/page-settings/settings' );
+			document.config.panel.default_route = 'panel/page-settings/settings';
```
Comment (matipojo) [inline assets/dev/js/editor/components/settings/base/manager.js:222]: Move it the the server side. the JS shouldn't touch the server config.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/settings/base/manager.js (hunk 1 above)
Comment (iNewLegend) [reply assets/dev/js/editor/components/settings/base/manager.js:222]: Handled.

File: core/settings/manager.php [hunk 2]
```diff
@@ -150,6 +150,10 @@ public static function get_settings_managers_config() {
 		foreach ( self::$settings_managers as $name => $manager ) {
 			$settings_model = $manager->get_model_for_config();
 
+			if ( ! $settings_model ) {
```
Comment (kobizz) [inline core/settings/manager.php:153]: Is there a real case where there isn't a settings model?

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/settings/manager.php (hunk 2 above)
Comment (iNewLegend) [reply core/settings/manager.php:153]: There is no real case, and if you see commits, you will see that it's part of test commit, anyway we have issue with the tests.

**Pull request merged**